From 3a4683508dbd23a6e97f301e05a99d7004d75002 Mon Sep 17 00:00:00 2001
From: Michael Cronenworth <mike@cchtml.com>
Date: Thu, 14 Feb 2013 21:26:23 -0600
Subject: [PATCH 1/2] AE: Pulse Audio fix crash on player stop.

PA is calling back functions on stream destruction and may try
to read a free'd stream variable. (trac bug 13903)

Signed-off-by: Michael Cronenworth <mike@cchtml.com>
---
 xbmc/cores/AudioEngine/Engines/PulseAE/PulseAEStream.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/xbmc/cores/AudioEngine/Engines/PulseAE/PulseAEStream.cpp b/xbmc/cores/AudioEngine/Engines/PulseAE/PulseAEStream.cpp
index 2322b47..f80d4d8 100644
--- a/xbmc/cores/AudioEngine/Engines/PulseAE/PulseAEStream.cpp
+++ b/xbmc/cores/AudioEngine/Engines/PulseAE/PulseAEStream.cpp
@@ -252,6 +252,10 @@ void CPulseAEStream::Destroy()
 
   if (m_Stream)
   {
+    pa_stream_set_state_callback(m_Stream, NULL, NULL);
+    pa_stream_set_write_callback(m_Stream, NULL, NULL);
+    pa_stream_set_latency_update_callback(m_Stream, NULL, NULL);
+    pa_stream_set_underflow_callback(m_Stream, NULL, NULL);
     pa_stream_disconnect(m_Stream);
     pa_stream_unref(m_Stream);
     m_Stream = NULL;
-- 
1.7.11.7

